src/charm.py

```
@@ -213,74 +146,70 @@ def get_manifest(self):
-    def _check_deployed_resources(self, manifest=None):
-        """Check the status of deployed resources, returning True if ok else raising CheckFailedError
+        # Using restart due to https://github.com/canonical/dex-auth-operator/issues/63
+        self._container.restart(self._container_name)
 
-        All abnormalities are captured in logs
+    def main(self, event):
+        try:
+            self._check_leader()
+        except CheckFailedError as err:
+            self.model.unit.status = err.status
+            return
 
-        Params:
-          manifest: (Optional) list of lightkube objects describing the entire application.  If
-                    omitted, will be computed using self.get_manifest()
-        """
-        if manifest:
-            expected_resources = manifest
-        else:
-            expected_resources = self.get_manifest()
-        found_resources = [None] * len(expected_resources)
-        errors = []
+        self.model.unit.status = MaintenanceStatus("Configuring dex charm")
+        self.ensure_state()
 
-        client = Client()
+        self._update_layer()
 
-        self.logger.info("Checking for expected resources")
-        for i, resource in enumerate(expected_resources):
-            try:
-                found_resources[i] = client.get(
-                    type(resource),
-                    resource.metadata.name,
-                    namespace=resource.metadata.namespace,
-                )
-            except ApiError:
-                errors.append(
-                    f"Cannot find k8s object for metadata '{resource.metadata}'"
-                )
-
-        self.logger.info("Checking readiness of found StatefulSets/Deployments")
-        statefulsets_ok, statefulsets_errors = validate_statefulsets_and_deployments(
-            found_resources
+        try:
+            self.handle_ingress()
+        except Exception as err:
+            self.model.unit.status = BlockedStatus(str(err))
+            return
```

Should we just let these `Exception`s raise to Juju?  It feels odd to catch **all** exceptions here and set them to Blocked.  I think the usual practice would be to set Waiting/Blocked when we know what happened, but to let exceptions raise (and the program error out) if we don't know (resulting in an `ErrorStatus` from Juju)